proxyopts/proxyopts.go

```diff
@@ -0,0 +1,97 @@
+package proxyopts
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+type javaOption struct {
+	Key   string
+	Value string
+}
+
+type javaOptions []javaOption
+
+func (o javaOption) Format() string {
+	return fmt.Sprintf("-D%s=%s", o.Key, o.Value)
+}
+
+func newJavaOption(key string, value string) javaOption {
+	return javaOption{
+		Key:   key,
+		Value: value,
+	}
+}
+
+func (o javaOptions) Format() string {
+	s := make([]string, len(o))
+	for i, opt := range o {
+		s[i] = opt.Format()
+	}
+	return strings.Join(s, " ")
+}
+
+func httpOpts(flags javaOptions, proxyURL string) (javaOptions, error) {
+	if len(proxyURL) == 0 {
+		return flags, nil
+	}
+
+	u, err := url.Parse(proxyURL)
+	if err != nil {
+		return flags, err
+	}
+
+	if len(u.Hostname()) == 0 || len(u.Port()) == 0 {
+		return flags, fmt.Errorf("if specifying a proxy URL, both hostname and port is required")
+	}
+
+	flags = append(flags, newJavaOption("http.proxyHost", u.Hostname()))
+	flags = append(flags, newJavaOption("https.proxyHost", u.Hostname()))
+	flags = append(flags, newJavaOption("http.proxyPort", u.Port()))
+	flags = append(flags, newJavaOption("https.proxyPort", u.Port()))
+
+	return flags, nil
+}
+
+// mangleWildcard takes a list of hostnames and prepends '*' if the hostname
+// starts with '.', then returns a new slice with the modified hostnames.
+func mangleWildcard(hosts []string) []string {
```

Comment:
Nit: Mix of english and norwegian.

Reply:
That's also an English word.

`mangle` v. lemleste, sønderrive, rive sund